static-instrumenter/maven-plugin/src/main/java/io/opentelemetry/contrib/staticinstrumenter/plugin/maven/InstrumentationAgent.java

```diff
@@ -0,0 +1,125 @@
+
+public class InstrumentationAgent {
+
+  // TODO: change after the new static-instr capable agent is released
+  public static final String JAR_FILE_NAME = "opentelemetry-javaagent.jar";
+  // TODO: change after the new static-instr capable agent is released
+  public static final String MAIN_CLASS = "io.opentelemetry.javaagent.StaticInstrumenter";
+
+  private static final Logger logger = LoggerFactory.getLogger(InstrumentationAgent.class);
+
+  private final String agentPath;
+  private final JarFile agentJar;
+
+  private InstrumentationAgent(String agentPath) throws IOException {
+    this.agentPath = agentPath;
+    this.agentJar = new JarFile(agentPath);
+  }
+
+  /**
+   * Creates new instance using {@value #JAR_FILE_NAME} classpath agent. Agent is copied to the
+   * target folder.
+   */
+  public static InstrumentationAgent createFromClasspathAgent(Path targetFolder)
+      throws IOException {
+
+    Path targetPath = targetFolder.resolve(JAR_FILE_NAME);
+    InputStream agentSource =
+        InstrumentationAgent.class.getClassLoader().getResourceAsStream(JAR_FILE_NAME);
+    if (agentSource == null) {
+      throw new IllegalStateException(
+          "Instrumented OTel agent not found in class path and JAR name: " + JAR_FILE_NAME);
+    }
+    try {
+      Files.copy(agentSource, targetPath, StandardCopyOption.REPLACE_EXISTING);
+    } catch (IOException exception) {
+      logger.error("Couldn't copy agent JAR using class resource: {}.", JAR_FILE_NAME);
+      throw exception;
+    }
+    return new InstrumentationAgent(targetPath.toString());
+  }
+
+  /**
+   * Returns the ProcessBuilder for instrumentation process.
+   *
+   * @param classpath classpath for instrumentation process (list of JARs to instrument, separated
+   *     with path separator)
+   * @param outputFolder output folder
+   * @return ProcessBuilder for instrumentation process
+   */
+  public ProcessBuilder getInstrumentationProcess(String classpath, Path outputFolder) {
+    return new ProcessBuilder(
+        "java",
+        "-Dotel.instrumentation.internal-class-loader.enabled=false",
+        String.format("-javaagent:%s", agentPath),
+        "-cp",
+        classpath,
+        MAIN_CLASS,
+        outputFolder.toString());
+  }
+
+  /**
+   * Adds classes from OpenTelemetry javaagent JAR to target file. Removes the shading and replaces
+   * classdata file extension with class file extension.
+   */
+  public void copyAgentClassesTo(Path targetFile, PackagingSupport packagingSupport)
+      throws IOException {
+
+    Set<String> existing = allNames(targetFile);
+    try (FileSystem targetFs = FileSystems.newFileSystem(targetFile, null)) {
+      consumeEntries(
+          agentJar,
+          (entry) -> {
+            String entryName = entry.getName();
+            if (isInstrumentationClass(entry)) {
+              String modifiedName = modifyAgentEntryName(entryName, packagingSupport);
+              if (!existing.contains(modifiedName)) {
```

Comment:
U sure about that? 
Current algo is to avoid moving entries from agent jar to target jar if they already exist. Hence:
1. We put all existing target jar entries in `existing` set
2. For each agent jar entry we check if it already exists in the target jar, if not - entry content is copied to target jar

If we switch contains() to add() we won't protect target jar entries from overwrite with agent entries.

Reply:
Hmm, true, looks like I got confused and was wrong. Nevermind then, thanks for being vigil 😛